sway-ir/src/optimize/simplify_cfg.rs

```diff
@@ -130,20 +140,9 @@ fn merge_blocks(context: &mut Context, function: &Function) -> Result<bool, IrEr
     // `from_block`.
     let succs = context.blocks[final_to_block.0]
         .successors(context)
-        .collect::<Vec<crate::block::Block>>();
+        .collect::<Vec<_>>();
```

Comment:
There are quite a few places throughout `sway-ir` where this is going on and it looks like the `Vec` and/or `.collect()` could be omitted.  But we're often passing the `Context` around as a mutable reference and so the borrow checker isn't happy with e.g., iterating over `.successors()` and `.update_phi_source_block()`  in the single expression.